### PR TITLE
feat: Use proxy wiring for event hasher

### LINKED ISSUE #12099: Use proxy wiring for `EventHasher`
- use the new proxy wiring for the `EventHasher`
- Extract an `EventHasher` interface to be compatible

### PR DESCRIPTION
**Pull request by alittley**
- closes #12099

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/EventHasher.java [hunk 1]
```diff
@@ -16,35 +16,20 @@
...
      * @param event the event to hash
      * @return the hashed event
      */
-    public GossipEvent hashEvent(@NonNull final GossipEvent event) {
-        cryptography.digestSync(event.getHashedData());
-        event.buildDescriptor();
-        return event;
-    }
+    @InputWireLabel("unhashed event")
+    GossipEvent hashEvent(@NonNull GossipEvent event);
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/EventHasher.java:34]:
```suggestion
    @NonNull
    GossipEvent hashEvent(@NonNull GossipEvent event);
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/DefaultEventHasher.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    public DefaultEventHasher(@NonNull final PlatformContext platformContext) {
+        this.cryptography = platformContext.getCryptography();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public GossipEvent hashEvent(@NonNull final GossipEvent event) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/DefaultEventHasher.java:44]:
```suggestion
    @NonNull
    public GossipEvent hashEvent(@NonNull final GossipEvent event) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/EventHasher.java (hunk 1 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/EventHasher.java:35]: Done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/DefaultEventHasher.java (hunk 2 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/DefaultEventHasher.java:45]: Done

**Review by cody-littley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**